include/crocoddyl_msgs/inertial_parameters_subscriber.h

```diff
@@ -0,0 +1,114 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2020-2023, Heriot-Watt University, University of Oxford
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_SUBSCRIBER_H_
+#define CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_SUBSCRIBER_H_
+
+#include "crocoddyl_msgs/MultibodyInertialParameters.h"
+
+#include <mutex>
+#include <ros/node_handle.h>
+
+typedef crocoddyl_msgs::MultibodyInertialParameters MultibodyInertialParameters;
+typedef const crocoddyl_msgs::MultibodyInertialParameters::ConstPtr
+    &MultibodyInertialParametersSharedPtr;
+namespace crocoddyl_msgs {
+
+
+class MultibodyInertialParametersRosSubscriber {
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  /**
+   * @brief Initialize the multi-body inertial parameters subscriber.
+   *
+   * @param[in] n_bodies  Number of bodies
+   * @param[in] topic     Topic name
+   */
+  MultibodyInertialParametersRosSubscriber(
+      const unsigned int n_bodies,
+      const std::string &topic = "/robot/multibody_inertial_parameters"):
+      n_bodies_(n_bodies), spinner_(2), has_new_msg_(false), is_processing_msg_(false), last_msg_time_(0.) {
+    ros::NodeHandle n;
+    sub_ = n.subscribe<MultibodyInertialParameters>(
+        topic, 1, &MultibodyInertialParametersRosSubscriber::callback, this,
+        ros::TransportHints().tcpNoDelay());
+    parameters.resize(10);
+    spinner_.start();
+    ROS_INFO_STREAM("Subscribing MultibodyInertialParameters messages on " << topic);
+  }
+
+  ~MultibodyInertialParametersRosSubscriber() = default;
+
+  /**
+   * @brief Get the latest inertial parameters
+   *
+   * @return  An std mape with the pair of body name, inertial parameters
+   */
+  
+  std::map<std::string, Eigen::VectorXd>
+  get_inertial_parameters() {
```

Comment:
1. Rename this function.
2. Adapt return type as justified above
```suggestion
  std::map<std::string, Vector10d>
  get_parameters() {
```

Please adapt the Python bindings as well. I won't mark this.